sync/src/orphan_block_pool.rs

```diff
@@ -43,6 +55,7 @@ impl OrphanBlockPool {
                 let (hashes, blocks): (Vec<_>, Vec<_>) = orphaned.into_iter().unzip();
                 for hash in hashes.iter() {
                     self.parents.remove(hash);
+                    self.leaders.remove(hash);
```

Comment:
In line 36, we should have already removed the block from leaders.

If we want to be protective, please add error logs if the hash does exist in leaders.